monai/handlers/checkpoint_loader.py

```diff
@@ -92,15 +105,22 @@ def __call__(self, engine: Engine) -> None:
         """
         checkpoint = torch.load(self.load_path, map_location=self.map_location)
 
-        if not self.strict_shape:
-            k, _ = list(self.load_dict.items())[0]
-            # single object and checkpoint is directly a state_dict
-            if len(self.load_dict) == 1 and k not in checkpoint:
-                checkpoint = {k: checkpoint}
+        k, _ = list(self.load_dict.items())[0]
+        # single object and checkpoint is directly a state_dict
+        if len(self.load_dict) == 1 and k not in checkpoint:
+            checkpoint = {k: checkpoint}
 
-            # skip items that don't match data shape
+        if not self.strict_shape:
+            pop_items: List[str] = []
             for k, obj in self.load_dict.items():
-                checkpoint[k] = copy_model_state(obj, checkpoint, inplace=False)[0]
+                if isinstance(obj, (torch.nn.Module)):
+                    # skip items that don't match key name or data shape
+                    checkpoint[k] = copy_model_state(obj, checkpoint, inplace=False)[0]
+                else:
+                    warnings.warn("`strict_shape` is False, load checkpoint for model, skip others in `load_dict`.")
```

Comment:
so, is this warning message correct here, should it be `strict` instead of `strict_shape`?

Reply:
Hi @wyli ,

As the `Note` in doc-string: "if `strict_shape=False`, will only load checkpoint for `torch.nn.Module` and skip other items in the `load_dict`."
We only skip others when `strict_shape` is False, optimizer, lr_scheduler, etc. can support `strict=False`, it's PyTorch behavior.

Thanks.

Reply:
yeah, `strict_shape=False` actually means ignore the `strict` flag, try to load all the nn.module.parameters from the checkpoint as long as the shapes match and keys match, and quietly ignore any mismatched keys, mismatched shapes... to me this is a confusing option

Reply:
Sure, let's discuss it in the other thread.
Thanks.